ambry-account/src/main/java/com/github/ambry/account/InMemoryUnknownAccountService.java

```diff
@@ -20,18 +20,20 @@
 
 /**
  * An implementation of {@link AccountService} that always has a single entry {@link Account#UNKNOWN_ACCOUNT}. Any
- * queries to this account service will unconditionally return {@link Account#UNKNOWN_ACCOUNT}. This account service
- * is in memory, and does not talk to any persistent storage service.
+ * queries by account name to this account service will unconditionally return {@link Account#UNKNOWN_ACCOUNT}. This
+ * account service is in memory, and does not talk to any persistent storage service.
  */
 class InMemoryUnknownAccountService implements AccountService {
+  private static final short CONTAINER_ID_FOR_LEGACY_PUT_PUBLIC_BLOB = -1;
+  private static final short CONTAINER_ID_FOR_LEGACY_PUT_PRIVATE_BLOB = -1;
   private static final Collection<Account> accounts =
       Collections.unmodifiableCollection(Collections.singletonList(Account.UNKNOWN_ACCOUNT));
   private volatile boolean isOpen = true;
 
   @Override
   public Account getAccountById(short accountId) {
     checkOpen();
-    return Account.UNKNOWN_ACCOUNT;
+    return accountId == Account.UNKNOWN_ACCOUNT_ID ? Account.UNKNOWN_ACCOUNT : null;
```

Comment:
Sorry to bring this up again. Can we call it LEGACY_ACCOUNT rather than UNKNOWN_ACCOUNT. There is a chance of confusion between UNKNOWN and INVALID. (Same comment for Container too)